packages/flutter_test/lib/src/matchers.dart

```diff
@@ -457,6 +459,36 @@ Matcher matchesSemanticsData({
   );
 }
 
+/// Asserts that the currently rendered widget meets the provided accessibility
+/// `guideline`.
+///
+/// This matcher requires the result to be awaited and for semantics to be
+/// enabled first.
+///
+/// ## Sample code
+///
+/// ```dart
+/// final SemanticsHandle handle = tester.ensureSemantics();
+/// await meetsGuideline(tester, meetsGuideline(textContrastGuideline));
+/// ```
+///
+/// Supported accessibility guidelines:
+///
+///   * [androidTapTargetGuideline]
+///   * [iOSTapTargetGuideline]
+///   * [textContrastGuideline]
+AsyncMatcher meetsGuideline(AccessibilityGuideline guideline) {
+  return new _MatchesAccessibilityGuideline(guideline);
+}
+
+/// The inverse matcher of [meetsGuideline].
+///
+/// This is needed because the [isNot] matcher does not compose with an
```

Comment:
sad. Maybe there should be an upstream isNotAsync matcher?

Reply:
There are, but they don't really mesh with how we handle async in widget tests, hence the AsyncMatcher work around.  I'm not really sure what the solution is here, maybe just make the guideline classes public so they can be unit tested?

Reply:
is there a bug filed on the `test` package about this limitation?

Reply:
Filled https://github.com/dart-lang/test/issues/923